numpy/core/src/private/ufunc_override.c

```diff
@@ -0,0 +1,532 @@
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+#define NO_IMPORT_ARRAY
+
+#include "npy_pycompat.h"
+#include "numpy/ufuncobject.h"
+#include "get_attr_string.h"
+#include "npy_import.h"
+
+#include "ufunc_override.h"
+
+static void
+normalize___call___args(PyUFuncObject *ufunc, PyObject *args,
+                    PyObject **normal_args, PyObject **normal_kwds,
+                    int nin)
+{
+    /* ufunc.__call__(*args, **kwds) */
+    int i;
+    int not_all_none;
+    int nargs = PyTuple_GET_SIZE(args);
+    PyObject *obj = PyDict_GetItemString(*normal_kwds, "sig");
+
+    /* ufuncs accept 'sig' or 'signature' normalize to 'signature' */
+    if (obj != NULL) {
+        Py_INCREF(obj);
+        PyDict_SetItemString(*normal_kwds, "signature", obj);
+        PyDict_DelItemString(*normal_kwds, "sig");
+    }
+
+    *normal_args = PyTuple_GetSlice(args, 0, nin);
+
+    /* If we have more args than nin, they must be the output variables.*/
```

Comment:
My sense would be that a check probably should happen, but maybe more logically in `ufunc_generic_call` (in https://github.com/numpy/numpy/blob/master/numpy/core/src/umath/ufunc_object.c#L4354) -- this particular path only gets called for overrides, and one might as well do it generically (the multiple layers of argument parsing will need some separation...)